frontend/src/reducers/productReducers.js

```diff
@@ -0,0 +1,27 @@
+import {
+  ALL_PRODUCTS_REQUEST,
+  ALL_PRODUCTS_SUCCESS,
+  ALL_PRODUCTS_FAIL,
+} from "constants/productContants";
+
+export const productReducer = (state = { products: [] }, action) => {
+  switch (action.type) {
+    case ALL_PRODUCTS_REQUEST:
+      return {
+        loading: true,
+        products: [],
```

Comment:
don't understood why u have de-structured only products and not loading and using loading without `state.loading` if it is working fine then u don't need to de-structure it

Reply:
Soryy but I am not destructing the product.

Reply:
It is just returning the empty product and loading to true.